src/Filter/StringFilter.php

```diff
@@ -87,14 +100,21 @@ public function filter(BaseProxyQueryInterface $query, $alias, $field, $data)
         // c.name > '1' => c.name OPERATOR :FIELDNAME
         $parameterName = $this->getNewParameterName($query);
 
-        $or = $query->getQueryBuilder()->expr()->orX();
+        $caseSensitive = $this->getOption('case_sensitive');
 
-        if ($this->getOption('case_sensitive')) {
-            $or->add(sprintf('%s.%s %s :%s', $alias, $field, $operator, $parameterName));
+        if (false === $caseSensitive) {
+            $clause = 'LOWER(%s.%s) %s :%s';
+        } elseif (true === $caseSensitive && null !== $this->doctrineConfig && Binary::class === $this->doctrineConfig->getCustomStringFunction('binary')) {
```

Comment:
Will this work if you don't have `beberlei/doctrineextensions` installed? I would expect some kind of exception.

Reply:
The condition will work, it returns false in that case, allowing the execution flow to reach the body of the `else` condition.